File: MirrorOfBinaryTree/MirrorOfBinaryTree.cpp
/*
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(NULL), right(NULL) {
	}
};*/
class Solution {
public:
    void Mirror(TreeNode *root) {
        if (root ==NULL)
            return;
        //交换根节点的左右结点
        TreeNode* temp = root->left;
        root->left = root->right;
        root->right =temp;
        
        //递归左子树
        if (root->left)
            Mirror(root->left);
        
        //递归右子树
        if (root->right)
            Mirror(root->right);
    }
};

File: MirrorOfBinaryTree/MirrorOfBinaryTree.py
# -*- coding:utf-8 -*-
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:
    # 返回镜像树的根节点
    def Mirror(self, root):
        # write code here
        if root == None:
            return
        temp = root.left
        root.left = root.right
        root.right = temp
        if root.left:
            self.Mirror(root.left)
        if root.right:
            self.Mirror(root.right)
